// office-ui-fabric-react/lib-amd/components/Callout/Callout.doc.js
define(["require", "exports", "react", "./examples/Callout.Basic.Example", "./examples/Callout.Nested.Example", "./examples/Callout.FocusTrap.Example", "./examples/Callout.Directional.Example", "./examples/Callout.Cover.Example"], function (require, exports, React, Callout_Basic_Example_1, Callout_Nested_Example_1, Callout_FocusTrap_Example_1, Callout_Directional_Example_1, Callout_Cover_Example_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var CalloutBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Callout/examples/Callout.Basic.Example.tsx');
    var CalloutNestedExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Callout/examples/Callout.Nested.Example.tsx');
    var CalloutFocusTrapExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Callout/examples/Callout.FocusTrap.Example.tsx');
    var CalloutDirectionalExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Callout/examples/Callout.Directional.Example.tsx');
    var CalloutCoverExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Callout/examples/Callout.Cover.Example.tsx');
    exports.CalloutPageProps = {
        title: 'Callout',
        componentName: 'Callout',
        componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Callout',
        examples: [
            {
                title: 'Default Callout',
                code: CalloutBasicExampleCode,
                view: React.createElement(Callout_Basic_Example_1.CalloutBasicExample, null)
            },
            {
                title: 'Nested Callout... Callout with a commandbar with a sub menu',
                code: CalloutNestedExampleCode,
                view: React.createElement(Callout_Nested_Example_1.CalloutNestedExample, null)
            },
            {
                title: 'Focus Trap Callout',
                code: CalloutFocusTrapExampleCode,
                view: React.createElement(Callout_FocusTrap_Example_1.CalloutFocusTrapExample, null)
            },
            {
                title: 'Callout with directional hint',
                code: CalloutDirectionalExampleCode,
                view: React.createElement(Callout_Directional_Example_1.CalloutDirectionalExample, null)
            },
            {
                title: 'Callout with cover',
                code: CalloutCoverExampleCode,
                view: React.createElement(Callout_Cover_Example_1.CalloutCoverExample, null)
            }
        ],
        overview: require('!raw-loader!office-ui-fabric-react/src/components/Callout/docs/CalloutOverview.md'),
        bestPractices: '',
        dos: require('!raw-loader!office-ui-fabric-react/src/components/Callout/docs/CalloutDos.md'),
        donts: require('!raw-loader!office-ui-fabric-react/src/components/Callout/docs/CalloutDonts.md'),
        isHeaderVisible: true,
        isFeedbackVisible: true
    };
});
