// office-ui-fabric-react/lib-amd/components/CommandBar/CommandBar.styles.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var COMMAND_BAR_HEIGHT = 44;
    exports.getStyles = function (props) {
        var className = props.className, theme = props.theme;
        var palette = theme.palette;
        return {
            root: [
                theme.fonts.medium,
                'ms-CommandBar',
                {
                    display: 'flex',
                    backgroundColor: palette.white,
                    padding: '0 14px 0 24px',
                    height: COMMAND_BAR_HEIGHT
                },
                className
            ],
            primarySet: [
                'ms-CommandBar-primaryCommand',
                {
                    flexGrow: '1',
                    display: 'flex',
                    alignItems: 'stretch'
                }
            ],
            secondarySet: [
                'ms-CommandBar-secondaryCommand',
                {
                    flexShrink: '0',
                    display: 'flex',
                    alignItems: 'stretch'
                }
            ]
        };
    };
});
