// office-ui-fabric-react/lib-commonjs/components/ColorPicker/examples/ColorPicker.Basic.Example.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var index_1 = require("office-ui-fabric-react/lib/index");
var Styling_1 = require("office-ui-fabric-react/lib/Styling");
var classNames = Styling_1.mergeStyleSets({
    wrapper: {
        display: 'flex'
    },
    column2: {
        marginLeft: 10
    },
    colorSquare: {
        width: 100,
        height: 100,
        margin: '16px 0',
        border: '1px solid #c8c6c4',
        selectors: (_a = {},
            _a[Styling_1.HighContrastSelector] = {
                MsHighContrastAdjust: 'none'
            },
            _a)
    }
});
var colorPickerStyles = {
    panel: { padding: 12 },
    root: {
        maxWidth: 352,
        minWidth: 352
    },
    colorRectangle: { height: 268 }
};
var ColorPickerBasicExample = /** @class */ (function (_super) {
    tslib_1.__extends(ColorPickerBasicExample, _super);
    function ColorPickerBasicExample() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = {
            color: index_1.getColorFromString('#ffffff'),
            alphaSliderHidden: false,
            showPreview: true
        };
        _this._updateColor = function (ev, colorObj) {
            _this.setState({ color: colorObj });
        };
        _this._onHideAlphaClick = function (ev, checked) {
            var color = _this.state.color;
            if (checked) {
                // If hiding the alpha slider, remove transparency from the color
                color = index_1.updateA(_this.state.color, 100);
            }
            _this.setState({ alphaSliderHidden: !!checked, color: color });
        };
        _this._onShowPreviewBoxClick = function (ev, checked) {
            _this.setState({ showPreview: !!checked });
        };
        return _this;
    }
    ColorPickerBasicExample.prototype.render = function () {
        var _a = this.state, color = _a.color, alphaSliderHidden = _a.alphaSliderHidden, showPreview = _a.showPreview;
        return (React.createElement("div", { className: classNames.wrapper },
            React.createElement(index_1.ColorPicker, { color: color, onChange: this._updateColor, alphaSliderHidden: alphaSliderHidden, showPreview: showPreview, styles: colorPickerStyles }),
            React.createElement("div", { className: classNames.column2 },
                React.createElement(index_1.Toggle, { label: "Hide alpha slider", onChange: this._onHideAlphaClick, checked: alphaSliderHidden }),
                React.createElement(index_1.Toggle, { label: "Show Preview Box", onChange: this._onShowPreviewBoxClick, checked: showPreview }))));
    };
    return ColorPickerBasicExample;
}(React.Component));
exports.ColorPickerBasicExample = ColorPickerBasicExample;
