// office-ui-fabric-react/lib/components/Breadcrumb/Breadcrumb.doc.js
import * as React from 'react';
import { BreadcrumbBasicExample } from './examples/Breadcrumb.Basic.Example';
import { BreadcrumbStaticExample } from './examples/Breadcrumb.Static.Example';
var BreadcrumbBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx');
var BreadcrumbBasicExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx');
var BreadcrumbStaticExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Static.Example.tsx');
var BreadcrumbStaticExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Static.Example.tsx');
export var BreadcrumbPageProps = {
    title: 'Breadcrumb',
    componentName: 'Breadcrumb',
    componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Breadcrumb',
    examples: [
        {
            title: 'Default Breadcrumb',
            code: BreadcrumbBasicExampleCode,
            codepenJS: BreadcrumbBasicExampleCodepen,
            view: React.createElement(BreadcrumbBasicExample, null)
        },
        {
            title: 'Breadcrumb with static width ',
            code: BreadcrumbStaticExampleCode,
            view: React.createElement(BreadcrumbStaticExample, null),
            codepenJS: BreadcrumbStaticExampleCodepen
        }
    ],
    overview: require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/docs/BreadcrumbOverview.md'),
    bestPractices: '',
    dos: require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/docs/BreadcrumbDos.md'),
    donts: require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/docs/BreadcrumbDonts.md'),
    isHeaderVisible: true,
    isFeedbackVisible: true
};
