// office-ui-fabric-react/lib/components/Persona/examples/Persona.Colors.Example.js
import * as tslib_1 from "tslib";
import * as React from 'react';
import { Persona, PersonaInitialsColor, PersonaSize } from 'office-ui-fabric-react/lib/Persona';
import { Stack } from 'office-ui-fabric-react/lib/Stack';
var sharedPersonaProps = {
    size: PersonaSize.size100,
    styles: {
        root: {
            width: 300,
            margin: 5
        }
    }
};
export var PersonaColorsExample = function () {
    return (React.createElement(Stack, { horizontal: true, wrap: true },
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.green }, sharedPersonaProps, { text: "green10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.darkGreen }, sharedPersonaProps, { text: "darkGreen20" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.teal }, sharedPersonaProps, { text: "teal10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.cyan }, sharedPersonaProps, { text: "cyan30" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.lightBlue }, sharedPersonaProps, { text: "lightBlue30" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.blue }, sharedPersonaProps, { text: "blue20" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.darkBlue }, sharedPersonaProps, { text: "darkBlue10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.violet }, sharedPersonaProps, { text: "violet10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.purple }, sharedPersonaProps, { text: "purple10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.magenta }, sharedPersonaProps, { text: "magenta10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.lightPink }, sharedPersonaProps, { text: "lightPink10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.pink }, sharedPersonaProps, { text: "pink10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.burgundy }, sharedPersonaProps, { text: "pinkRed10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.lightRed }, sharedPersonaProps, { text: "red10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.darkRed }, sharedPersonaProps, { text: "darkRed20" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.orange }, sharedPersonaProps, { text: "orange10" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.rust }, sharedPersonaProps, { text: "orange30" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.gold }, sharedPersonaProps, { text: "orangeYellow20" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.warmGray }, sharedPersonaProps, { text: "gray30" })),
        React.createElement(Persona, tslib_1.__assign({ initialsColor: PersonaInitialsColor.coolGray }, sharedPersonaProps, { text: "gray20" }))));
};
