// @uifabric/example-data/lib-amd/listItems.js
define(["require", "exports", "./lorem"], function (require, exports, lorem_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var DATA = {
        color: ['red', 'blue', 'green', 'yellow'],
        shape: ['circle', 'square', 'triangle'],
        location: ['Seattle', 'New York', 'Chicago', 'Los Angeles', 'Portland']
    };
    /** @internal */
    function createListItems(count, startIndex) {
        if (startIndex === void 0) { startIndex = 0; }
        return Array.apply(null, Array(count)).map(function (item, index) {
            var size = 150 + Math.round(Math.random() * 100);
            return {
                thumbnail: "//placehold.it/" + size + "x" + size,
                key: 'item-' + (index + startIndex) + ' ' + lorem_1.lorem(4),
                name: lorem_1.lorem(5),
                description: lorem_1.lorem(10 + Math.round(Math.random() * 50)),
                color: _randWord(DATA.color),
                shape: _randWord(DATA.shape),
                location: _randWord(DATA.location),
                width: size,
                height: size
            };
        });
    }
    exports.createListItems = createListItems;
    /** @internal */
    function createGroups(groupCount, groupDepth, startIndex, itemsPerGroup, level, key, isCollapsed) {
        if (level === void 0) { level = 0; }
        if (key === void 0) { key = ''; }
        if (key !== '') {
            key = key + '-';
        }
        var count = Math.pow(itemsPerGroup, groupDepth);
        return Array.apply(null, Array(groupCount)).map(function (value, index) {
            return {
                count: count,
                key: 'group' + key + index,
                name: 'group ' + key + index,
                startIndex: index * count + startIndex,
                level: level,
                isCollapsed: isCollapsed,
                children: groupDepth > 1 ? createGroups(groupCount, groupDepth - 1, index * count + startIndex, itemsPerGroup, level + 1, key + index) : []
            };
        });
    }
    exports.createGroups = createGroups;
    /** @internal */
    function isGroupable(key) {
        return key === 'color' || key === 'shape' || key === 'location';
    }
    exports.isGroupable = isGroupable;
    function _randWord(array) {
        var index = Math.floor(Math.random() * array.length);
        return array[index];
    }
});
