// office-ui-fabric-react/lib-commonjs/components/GroupedList/GroupedList.doc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var GroupedList_Basic_Example_1 = require("./examples/GroupedList.Basic.Example");
var GroupedList_Custom_Example_1 = require("./examples/GroupedList.Custom.Example");
var GroupedListBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx');
var GroupedListCustomExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Custom.Example.tsx');
var GroupedListBasicExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx');
var GroupedListCustomExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Custom.Example.tsx');
exports.GroupedListPageProps = {
    title: 'GroupedList',
    componentName: 'GroupedList',
    componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/GroupedList',
    examples: [
        {
            title: 'GroupedList basic example',
            code: GroupedListBasicExampleCode,
            codepenJS: GroupedListBasicExampleCodepen,
            view: React.createElement(GroupedList_Basic_Example_1.GroupedListBasicExample, null)
        },
        {
            title: 'GroupedList example with custom header and footer',
            code: GroupedListCustomExampleCode,
            codepenJS: GroupedListCustomExampleCodepen,
            view: React.createElement(GroupedList_Custom_Example_1.GroupedListCustomExample, null)
        }
    ],
    overview: require('!raw-loader!office-ui-fabric-react/src/components/GroupedList/docs/GroupedListOverview.md'),
    bestPractices: '',
    dos: '',
    donts: '',
    isHeaderVisible: true,
    isFeedbackVisible: true
};
