// office-ui-fabric-react/lib/components/ChoiceGroup/ChoiceGroup.doc.js
import * as React from 'react';
import { ChoiceGroupBasicExample } from './examples/ChoiceGroup.Basic.Example';
import { ChoiceGroupLabelExample } from './examples/ChoiceGroup.Label.Example';
import { ChoiceGroupCustomExample } from './examples/ChoiceGroup.Custom.Example';
import { ChoiceGroupImageExample } from './examples/ChoiceGroup.Image.Example';
import { ChoiceGroupIconExample } from './examples/ChoiceGroup.Icon.Example';
var ChoiceGroupBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Basic.Example.tsx');
var ChoiceGroupBasicExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Basic.Example.tsx');
var ChoiceGroupLabelExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Label.Example.tsx');
var ChoiceGroupLabelExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Label.Example.tsx');
var ChoiceGroupCustomExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Custom.Example.tsx');
var ChoiceGroupCustomExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Custom.Example.tsx');
var ChoiceGroupImageExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Image.Example.tsx');
var ChoiceGroupImageExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Image.Example.tsx');
var ChoiceGroupIconExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Icon.Example.tsx');
var ChoiceGroupIconExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/ChoiceGroup/examples/ChoiceGroup.Icon.Example.tsx');
export var ChoiceGroupPageProps = {
    title: 'ChoiceGroup',
    componentName: 'ChoiceGroup',
    componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/ChoiceGroup',
    examples: [
        {
            title: 'Default ChoiceGroup',
            code: ChoiceGroupBasicExampleCode,
            view: React.createElement(ChoiceGroupBasicExample, null),
            codepenJS: ChoiceGroupBasicExampleCodepen
        },
        {
            title: 'ChoiceGroup with a custom label',
            code: ChoiceGroupLabelExampleCode,
            view: React.createElement(ChoiceGroupLabelExample, null),
            codepenJS: ChoiceGroupLabelExampleCodepen
        },
        {
            title: 'ChoiceGroup with dropdown',
            code: ChoiceGroupCustomExampleCode,
            codepenJS: ChoiceGroupCustomExampleCodepen,
            view: React.createElement(ChoiceGroupCustomExample, null)
        },
        {
            title: 'ChoiceGroups with images',
            code: ChoiceGroupImageExampleCode,
            codepenJS: ChoiceGroupImageExampleCodepen,
            view: React.createElement(ChoiceGroupImageExample, null)
        },
        {
            title: 'ChoiceGroup with icons',
            code: ChoiceGroupIconExampleCode,
            view: React.createElement(ChoiceGroupIconExample, null),
            codepenJS: ChoiceGroupIconExampleCodepen
        }
    ],
    overview: require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/docs/ChoiceGroupOverview.md'),
    bestPractices: '',
    dos: require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/docs/ChoiceGroupDos.md'),
    donts: require('!raw-loader!office-ui-fabric-react/src/components/ChoiceGroup/docs/ChoiceGroupDonts.md'),
    isHeaderVisible: true,
    isFeedbackVisible: true,
    allowNativeProps: true,
    nativePropsElement: 'input'
};
