// office-ui-fabric-react/lib-amd/components/Breadcrumb/Breadcrumb.doc.js
define(["require", "exports", "react", "./examples/Breadcrumb.Basic.Example", "./examples/Breadcrumb.Static.Example"], function (require, exports, React, Breadcrumb_Basic_Example_1, Breadcrumb_Static_Example_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var BreadcrumbBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx');
    var BreadcrumbBasicExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx');
    var BreadcrumbStaticExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Static.Example.tsx');
    var BreadcrumbStaticExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Static.Example.tsx');
    exports.BreadcrumbPageProps = {
        title: 'Breadcrumb',
        componentName: 'Breadcrumb',
        componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Breadcrumb',
        examples: [
            {
                title: 'Default Breadcrumb',
                code: BreadcrumbBasicExampleCode,
                codepenJS: BreadcrumbBasicExampleCodepen,
                view: React.createElement(Breadcrumb_Basic_Example_1.BreadcrumbBasicExample, null)
            },
            {
                title: 'Breadcrumb with static width ',
                code: BreadcrumbStaticExampleCode,
                view: React.createElement(Breadcrumb_Static_Example_1.BreadcrumbStaticExample, null),
                codepenJS: BreadcrumbStaticExampleCodepen
            }
        ],
        overview: require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/docs/BreadcrumbOverview.md'),
        bestPractices: '',
        dos: require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/docs/BreadcrumbDos.md'),
        donts: require('!raw-loader!office-ui-fabric-react/src/components/Breadcrumb/docs/BreadcrumbDonts.md'),
        isHeaderVisible: true,
        isFeedbackVisible: true
    };
});
