// office-ui-fabric-react/lib-commonjs/components/Checkbox/Checkbox.doc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Checkbox_Basic_Example_1 = require("./examples/Checkbox.Basic.Example");
var Checkbox_Other_Example_1 = require("./examples/Checkbox.Other.Example");
var CheckboxBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Checkbox/examples/Checkbox.Basic.Example.tsx');
var CheckboxBasicExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/Checkbox/examples/Checkbox.Basic.Example.tsx');
var CheckboxOtherExamplesCode = require('!raw-loader!office-ui-fabric-react/src/components/Checkbox/examples/Checkbox.Other.Example.tsx');
var CheckboxOtherExamplesCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/Checkbox/examples/Checkbox.Other.Example.tsx');
exports.CheckboxPageProps = {
    title: 'Checkbox',
    componentName: 'Checkbox',
    componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Checkbox',
    examples: [
        {
            title: 'Default Checkbox',
            code: CheckboxBasicExampleCode,
            view: React.createElement(Checkbox_Basic_Example_1.CheckboxBasicExample, null),
            codepenJS: CheckboxBasicExampleCodepen
        },
        {
            title: 'Implementation Examples',
            code: CheckboxOtherExamplesCode,
            view: React.createElement(Checkbox_Other_Example_1.CheckboxOtherExamples, null),
            codepenJS: CheckboxOtherExamplesCodepen
        }
    ],
    overview: require('!raw-loader!office-ui-fabric-react/src/components/Checkbox/docs/CheckboxOverview.md'),
    bestPractices: '',
    dos: require('!raw-loader!office-ui-fabric-react/src/components/Checkbox/docs/CheckboxDos.md'),
    donts: require('!raw-loader!office-ui-fabric-react/src/components/Checkbox/docs/CheckboxDonts.md'),
    isHeaderVisible: true,
    isFeedbackVisible: true
};
