// office-ui-fabric-react/lib-amd/components/pickers/Pickers.doc.js
define(["require", "exports", "react", "./examples/Picker.CustomResult.Example", "./examples/TagPicker.Basic.Example"], function (require, exports, React, Picker_CustomResult_Example_1, TagPicker_Basic_Example_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var TagPickerExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx');
    var TagPickerExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx');
    var PickerCustomResultExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/pickers/examples/Picker.CustomResult.Example.tsx');
    var PickerCustomResultExampleCodepen = require('!@uifabric/codepen-loader!office-ui-fabric-react/src/components/pickers/examples/Picker.CustomResult.Example.tsx');
    exports.PickersPageProps = {
        title: 'Pickers',
        componentName: 'Pickers',
        componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Pickers',
        examples: [
            {
                title: 'Tag Picker',
                code: TagPickerExampleCode,
                codepenJS: TagPickerExampleCodepen,
                view: React.createElement(TagPicker_Basic_Example_1.TagPickerBasicExample, null)
            },
            {
                title: 'Custom Picker (Document Picker)',
                code: PickerCustomResultExampleCode,
                codepenJS: PickerCustomResultExampleCodepen,
                view: React.createElement(Picker_CustomResult_Example_1.PickerCustomResultExample, null)
            }
        ],
        overview: require('!raw-loader!office-ui-fabric-react/src/components/pickers/docs/PickersOverview.md'),
        isHeaderVisible: true,
        isFeedbackVisible: true
    };
});
